pyccel/ast/cwrapper.py

```diff
@@ -114,7 +116,7 @@ def __init__(self, python_func_args, python_func_kwargs, c_func_args, parse_args
             raise TypeError('Python func args should be a Variable')
         if not isinstance(python_func_kwargs, Variable):
             raise TypeError('Python func kwargs should be a Variable')
-        if not isinstance(c_func_args, list) and any(not isinstance(c, Variable) for c in c_func_args):
+        if not isinstance(c_func_args, (tuple, Tuple)) or any(not isinstance(c, (Variable, FunctionAddress)) for c in c_func_args):
```

Comment:
`tuple` should be sufficient. `Tuple` is a sympy object which we should probably remove at some point. It's therefore better to not add additional unnecessary references to it

Reply:
I made a comment to the outdated code, because it seemed natural to continue the same discussion. sorry for the confusion

Reply:
My point there was that we probably don't even need to check the type of `c_func_args`...